aplib.net-demo/Assets/Scripts/LevelGeneration/LevelSpawner.cs

```diff
@@ -217,16 +214,27 @@ private static List<Teleporter.Teleporter> GetAllEntryTeleporters(GameObject tel
         /// <param name="direction">The direction in which the door should be placed.</param>
         /// <param name="startComponent">The component to start the search from.</param>
         /// <param name="parent">The parent of the door.</param>
-        private void PlaceDoor(Cell cell1, Cell cell2, Direction direction, ConnectedComponent startComponent,
+        private (int x, int z, int x2, int z2)? PlaceDoor(Cell cell1, Cell cell2, Direction direction,
+            ConnectedComponent startComponent,
             Transform parent)
         {
             List<Cell> emptyCells = GetEmptyCells(startComponent);
+
+            // If there are no empty cells, we cannot place a door. This can occur when there is exactly one cell per key and placed teleporters.
+            if (emptyCells.Count == 0) return null;
+
             Cell itemCell = emptyCells[SharedRandom.Next(emptyCells.Count)];
 
             if (cell1.Tile is Room room)
+            {
                 _gameObjectPlacer.PlaceDoorInDirection(cell1.X, cell1.Z, room, direction, itemCell, parent);
-            else if (cell2.Tile is Room room2)
-                _gameObjectPlacer.PlaceDoorInDirection(cell2.X, cell2.Z, room2, direction.Opposite(), itemCell, parent);
+                return (cell1.X, cell1.Z, cell2.X, cell2.Z);
+            }
+
+            if (cell2.Tile is not Room room2) return null;
+
+            _gameObjectPlacer.PlaceDoorInDirection(cell2.X, cell2.Z, room2, direction.Opposite(), itemCell, parent);
+            return (cell2.X, cell2.Z, cell1.X, cell1.Z);
```

Comment:
```suggestion
            if (cell1.Tile is Room room1)
            {
                _gameObjectPlacer.PlaceDoorInDirection(cell1.X, cell1.Z, room1, direction, itemCell, parent);
                return (cell1.X, cell1.Z, cell2.X, cell2.Z);
            }

            // ReSharper disable once InvertIf
            if (cell2.Tile is Room room2)
            {
                _gameObjectPlacer.PlaceDoorInDirection(cell2.X, cell2.Z, room2, direction.Opposite(), itemCell, parent);
                return (cell2.X, cell2.Z, cell1.X, cell1.Z);
            }

            return null;
```

Reply:
For better readability.

Reply:
I doesn't let me apply the changes because there is a deleted line involved. I have made and pushed the changes myself.